### PR TITLE
Update solid model documentation

### PR DESCRIPTION
**Pull request by joshua-white**
This PR re-organizes much of the Solid model documentation.  There are a few remaining items, though I am undecided whether we should do them in this PR or separate ones.  For the moment I am leaning towards merging and then addressing these is smaller PRs:

  - [ ] Describing the new porosity model hierarchy (@CusiniM, maybe better in a separate PR?)
  - [ ] Damage model documentation (will need help from @andrembcosta)
  - [ ] The triaxial driver and Delft-Egg documentation will need to be updated as we merge the Delft-Egg branch.

To review, we have this branch building on readthedocs.  Look [here.](https://geosx-geosx.readthedocs-hosted.com/en/docs-solidmodels/coreComponents/constitutive/docs/solid/SolidModels.html)

### CODE REVIEW

File: src/coreComponents/constitutive/docs/solid/Plasticity.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   
+in which :math:`\dot{\lambda} \geq 0` is the magnitude of plastic strain rate and :math:`g` is the plastic potential. The elastic strain is related to Cauchy stress tensor in rate form as:
+
+.. math::
+  \dot{\boldsymbol{\sigma}} = \tensor{c}^e \cdot \dot{\boldsymbol{\epsilon}}^e,
+
+where :math:`\tensor{c}^e` is the fourth order elastic stiffness tensor. The Cauchy stress tensor is related to the total strain as
+
+.. math::
+  \dot{\boldsymbol{\sigma}} = \boldsymbol{c}^{ep} \cdot \dot{\boldsymbol{\epsilon}},
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/Plasticity.rst:31]:
```suggestion
  \dot{\boldsymbol{\sigma}} = \boldsymbol{c}^{ep} : \dot{\boldsymbol{\epsilon}}, 
```

File: src/coreComponents/constitutive/docs/solid/Plasticity.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+The plastic strain tensor is obtained from the flow rule: 
+
+.. math::
+   \dot{\boldsymbol{\epsilon}}^p=\dot{\lambda}\frac{\partial g}{\partial\boldsymbol{\sigma}},
+   
+in which :math:`\dot{\lambda} \geq 0` is the magnitude of plastic strain rate and :math:`g` is the plastic potential. The elastic strain is related to Cauchy stress tensor in rate form as:
+
+.. math::
+  \dot{\boldsymbol{\sigma}} = \tensor{c}^e \cdot \dot{\boldsymbol{\epsilon}}^e,
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/Plasticity.rst:26]:
```suggestion
  \dot{\boldsymbol{\sigma}} = \tensor{c}^e : \dot{\boldsymbol{\epsilon}}^e,
```

File: src/coreComponents/constitutive/docs/solid/ElasticIsotropic.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+This model may be used for solid materials with a linear elastic isotropic behavior.
+The relationship between stress and strain is given by `Hooke's Law <https://en.wikipedia.org/wiki/Hooke%27s_law>`__,
+expressed as:
+
+.. math::
+   \sigma_{ij} = \lambda \epsilon_{kk} + 2 \mu \epsilon_{ij},
+
+where :math:`\sigma_{ij}` is the :math:`ij` component of the Cauchy stress tensor,
+:math:`\epsilon_{ij}` is the :math:`ij` component of the the strain tensor,
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/ElasticIsotropic.rst:18]:
```suggestion
:math:`\epsilon_{ij}` is the :math:`ij` component of the strain tensor,
```

File: src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+=========================
+
+This model may be used for solid materials with a pressure-dependent elastic isotropic behavior.
+The relationship between stress and strain is given by a `hyperelastic law <https://en.wikipedia.org/wiki/Hyperelastic_material>`__. The elastic constitutive equations for the volumetric and deviatoric stresses and strain are expressed as:
+
+.. math::
+   p = p_0 \exp{\left(\frac{\epsilon_{v0}-\epsilon_v^e}{C_r}\right)} \, , \quad q = 3 \mu \epsilon_s^e
+   
+where :math:`p` and  :math:`q` are the volumetric and deviatoric components of the Cauchy stress tensor.
+:math:`\epsilon_{v}^e` and :math:`\epsilon_{s}^e` are the volumetric and deviatoric components of the the strain tensor. :math:`\epsilon_{v0}` and :math:`p_0` are the initial volumetric strain and initial pressure. :math:`C_r` denotes the elastic compressibility index,
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst:17]:
```suggestion
:math:`\epsilon_{v}^e` and :math:`\epsilon_{s}^e` are the volumetric and deviatoric components of the strain tensor. :math:`\epsilon_{v0}` and :math:`p_0` are the initial volumetric strain and initial pressure. :math:`C_r` denotes the elastic compressibility index,
```

File: src/coreComponents/constitutive/docs/solid/DruckerPrager.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+.. figure:: DPyield.png
+   :align: center
+   :width: 500
+   :figclass: align-center
+
+   Mohr-Coulomb and Drucker-Prager yield surfaces in principal stress axes (Borja, 2002).
+
+The material behavior is linear elastic (see :ref:`ElasticIsotropic`) for :math:`f < 0`, and plastic for :math:`f =0`.
+The two material parameters :math:`a` and :math:`b` are derived by approximating the Mohr-Coulomb surface with a cone.
+Figure 1 shows the Mohr-Coulomb yield surface and circumscribing Drucker-Prager surface in principal stress space.
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/DruckerPrager.rst:25]:
```suggestion
Figure 3 shows the Mohr-Coulomb yield surface and circumscribing Drucker-Prager surface in principal stress space.
```

File: src/coreComponents/constitutive/docs/solid/DruckerPrager.rst [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+The Drucker-Prager yield surface has a circular cross-section in the deviatoric plane that passes through the tension or compression corners of the Mohr-Coulomb yield surface, as shown in the Figure 2.
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/DruckerPrager.rst:26]:
```suggestion
The Drucker-Prager yield surface has a circular cross-section in the deviatoric plane that passes through the tension or compression corners of the Mohr-Coulomb yield surface, as shown in the Figure 4.
```

File: src/coreComponents/constitutive/docs/solid/DruckerPrager.rst [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+The material parameters :math:`a` and :math:`b` are derived as:
+
+.. math::
+   a = \frac{6 \, c \, \cos\phi}{3 \pm \sin\phi} \, , \quad b=\frac{6  \, \sin\phi}{3 \pm \sin\phi}
+
+where plus signs are for circles passing through the tension corners, and minus signs are for circles passing through compression corners.
+Also, :math:`\phi` and :math:`c` denote friction angle and cohesion, respectively, as defined by the Mohr-Coulomb failure envelope shown in Figure 3.
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/DruckerPrager.rst:33]:
```suggestion
Also, :math:`\phi` and :math:`c` denote friction angle and cohesion, respectively, as defined by the Mohr-Coulomb failure envelope shown in Figure 5.
```

**Comment by CusiniM** [comment]
> This PR re-organizes much of the Solid model documentation. There are a few remaining items, though I am undecided whether we should do them in this PR or separate ones. For the moment I am leaning towards merging and then addressing these is smaller PRs:
> 
>     * [ ]  Describing the new porosity model hierarchy (@CusiniM, maybe better in a separate PR?)
> 
>     * [ ]  Damage model documentation (will need help from @andrembcosta)
> 
>     * [ ]  The triaxial driver and Delft-Egg documentation will need to be updated as we merge the Delft-Egg branch.
> 
> 
> To review, we have this branch building on readthedocs. Look [here.](https://geosx-geosx.readthedocs-hosted.com/en/docs-solidmodels/coreComponents/constitutive/docs/solid/SolidModels.html)

I would document how to use the compound constitutive models in a separate PR once we are fully done with them (e.g., once we solved issue #1528).

File: src/coreComponents/constitutive/docs/solid/DruckerPragerExtended.rst [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+.. math::
+   b = b_i + \frac{\lambda}{m+\lambda} \left( b_r - b_i \right)
+
+with :math:`m` a parameter controlling the hardening rate.  Here, :math:`b_r` is the residual yield surface slope.  
+If :math:`b_r < b_i`, hardening behavior will be observed, while for :math:`b_r < b_i` softening behavior will occur.
+
+In the resulting model, the yield surface begins at an initial position defined by the initial cohesion and friction angle.
+As plastic deformation occurs, the friction angle hardens (or softens) so that it asymptoptically approaches a
+residual friction angle.  The vertex of the cone remains fixed in p-q space, but the cohesion intercept evolves in
+tandem with the friction angle.  See Liu et al. (2020) for complete details.
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/DruckerPragerExtended.rst:33]:
```suggestion
tandem with the friction angle.  See `Liu et al. (2020) <https://doi.org/10.1007/s00603-019-01992-5>` for complete details.
```

File: src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+############################################
+
+Overview
+=========================
+
+This model may be used for solid materials with a pressure-dependent elastic isotropic behavior.
+The relationship between stress and strain is given by a `hyperelastic law <https://en.wikipedia.org/wiki/Hyperelastic_material>`__. The elastic constitutive equations for the volumetric and deviatoric stresses and strain are expressed as:
+
+.. math::
+   p = p_0 \exp{\left(\frac{\epsilon_{v0}-\epsilon_v^e}{C_r}\right)} \, , \quad q = 3 \mu \epsilon_s^e
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst:14]:
```suggestion
   p = p_0 \exp{\left(\frac{\epsilon_{v0}-\epsilon_v^e}{c_r}\right)} \, , \quad q = 3 \mu \epsilon_s^e
```

File: src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst (hunk 4 above)
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst:17]:
```suggestion
:math:`\epsilon_{v}^e` and :math:`\epsilon_{s}^e` are the volumetric and deviatoric components of the the strain tensor. :math:`\epsilon_{v0}` and :math:`p_0` are the initial volumetric strain and initial pressure. :math:`c_r` denotes the elastic compressibility index,
```

File: src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+   
+where :math:`p` and  :math:`q` are the volumetric and deviatoric components of the Cauchy stress tensor.
+:math:`\epsilon_{v}^e` and :math:`\epsilon_{s}^e` are the volumetric and deviatoric components of the the strain tensor. :math:`\epsilon_{v0}` and :math:`p_0` are the initial volumetric strain and initial pressure. :math:`C_r` denotes the elastic compressibility index,
+and :math:`\mu` is the elastic shear modulus. In this model, the shear modulus is constant and the bulk modulus, :math:`K`, varies linearly with pressure as follows: 
+
+.. math::
+   K = -\frac{p}{C_r}
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst:21]:
```suggestion
   K = -\frac{p}{c_r}
```

File: src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+Parameters
+=========================
+
+The following attributes are supported.  Note that two elastic constants :math:`C_r` and :math:`\mu`, as well as the initial volumetric strain and initial pressure need to be provided.  The "default" keyword in front of certain properties indicates that this
```
Comment (shabnamjs) [inline src/coreComponents/constitutive/docs/solid/ElasticIsotropicPressureDependent.rst:26]:
```suggestion
The following attributes are supported.  Note that two elastic constants :math:`c_r` and :math:`\mu`, as well as the initial volumetric strain and initial pressure need to be provided.  The "default" keyword in front of certain properties indicates that this
```

**Review by shabnamjs — APPROVED** [review]
Looks great to me. I just included a few minor edits.

**Pull request merged**